src/app.js

```diff
@@ -25,74 +53,38 @@ export const start = async args => {
 
   const username = program.user;
 
-  let reposList;
-
-  //TODO: prompt for username if none provided
   if (!username) {
-    console.error(
-      `Use option -u or --user to search repositories by Github use
+    process.stderr.write(
+      `Use option -u or --user to search repositories by Github user
       \nUsage: git-started -u jwu910`.trim(),
     );
 
     return;
   }
 
   try {
-    const response = await getRepos(username);
-
-    const repositories = response.user.repositories.edges;
-
-    if (!repositories.length) {
-      throw new Error('No repositories found.');
-    }
-
-    reposList = repositories.map(({ node }) => {
-      const { id, name, stargazers } = node;
-
-      let { description } = node;
-
-      const termWidth = process.stdout.columns;
-
-      if (description === null) {
-        description = 'No Description';
-      }
-
-      const limit = termWidth - name.length;
-
-      return {
-        title: `${name} - ${stargazers.totalCount} stars - ${truncate(
-          description,
-          limit - 20,
-          '...',
-        )}`,
-        value: {
-          id,
-          repo: name,
-          owner: username,
-        },
-      };
-    });
+    const userGraph = await getRepos(username);
+    const reposList = reposListFromGraphEdges(
+      userGraph.user.repositories.edges,
+      username,
+    );
 
-    let selection = await prompts({
+    const chosenRepo = await prompts({
       type: 'autocomplete',
       name: `repository`,
       message: `Select ${username}'s repository to fork or start typing to search`,
       choices: [...reposList],
     });
 
-    const fork = await forkRepo(selection);
-
-    process.stdout.write(`Cloning into ${cloneUrl}`);
-
-    const cloneUrl = program.ssh ? fork.data.ssh_url : fork.data.clone_url;
-
-    await cloneRepo(cloneUrl);
-  } catch (error) {
-    process.stderr.write(
-      'Unable to fork repository, check params and try again \n',
-      error,
-    );
+    if (!chosenRepo.repository) {
+      throw new Error('repository not found.');
+    }
 
+    const fork = await forkRepo(chosenRepo);
+    await cloneRepo(program.ssh ? fork.data.ssh_url : fork.data.clone_url);
+    process.stdout.write(`Successfully forked and cloned repo.\n`);
+  } catch (err) {
+    process.stderr.write(`${err}\n`);
```

Comment:
Couple comments about these lines

When I mentioned the modular try/catch blocks i had envisioned something like this:

```
try {
	const fork = await forkRepo(selection);

	console.log('Successfully forked repo');
}
catch (error) {
	console.error('Unable to fork repo');
	console.error('Reason: ', error);

	process.exit(1);
}

try {
	await cloneRepo(program.ssh ? fork.data.ssh_url : fork.data.clone_url);

	console.log('Successfully cloned repository');
}
catch (error) {
	console.error('Unable to clone repo');
	console.error('Reason: ', error);

	process.exit(1);
}

```

I'd have to see this snippet in a larger scope to see what the formatting looks like impelmenting the try/catches, but from my readings about the use of async/await over promise.then()'s uses modular try catches similar like this.

I'll try to do some research in the mean time, but let me know what you think about this

Reply:
This code actually doesn't run. `const` variables are block-scoped, so any code outside the try/catch block can't access them. That's why I had added a `var` earlier; `var`-declared variables are function-scoped, so they can be seen outside the try/catch where they are declared. AFAIK, there are two allternative ways of doing it:

One is what I had tried before, where I had split each try/catch into its own function. This way, I can return the primary "output" of the try/catch block.

Another way that I avoided goes like this:
```
let fork;
try {
    fork = await forkRepo();
    ...
}
```
This way, `fork` is explicitly scoped outside of the try/catch block. This seems like a more "ES6" way to do it. I didn't do it because I just don't like the way it looks. You can make the call, really, on what's the right way to do it.

Reply:
your comment about `let` is accurate. 

And we'll use that. You can read up more about var/const/let, but the most common patterns I've seen now is to avoid `var` all together with the introduction of `let/const`. 

But this comment was more focused on the structure of the `try/catches`. And this was just typed up as an example.

https://github.com/jwu910/git-started/pull/20/files/9c41ffa858ebe0a78b36176ea0def1bbfda74db3#r261891765 I made a comment further up about why I think separate try catches would be the way to go.

Reply:
Okay, I’ll go with *let*. That will let me restructure the try/catch blocks better.